rest/config.go

```diff
@@ -1023,9 +1023,22 @@ func RunServer(config *ServerConfig) {
 	}
 
 	sc := NewServerContext(config)
-	for _, dbConfig := range config.Databases {
+	bucketUUIDToDbs := make(map[string][]string, len(config.Databases))
+
+	for dbName, dbConfig := range config.Databases {
 		if _, err := sc.AddDatabaseFromConfig(dbConfig); err != nil {
-			base.Fatalf("Error opening database %s: %+v", base.MD(dbConfig.Name), err)
+			base.Fatalf("Error opening database %s: %+v", base.MD(dbName), err)
+		}
+		dbContext, err := sc.GetDatabase(dbName)
+		if err != nil {
+			base.Fatalf("Error getting database context %s: %+v", base.MD(dbName), err)
+		}
+		if uuid, err := dbContext.Bucket.UUID(); err == nil {
+			if value := bucketUUIDToDbs[uuid]; value != nil {
+				base.Warnf("Unsupported config: Database %q shares bucket %q with databases %s",
```

Comment:
This isn't unsupported, as discussed on the ticket (unsupported implies that if customers do this, support can/will refuse to work tickets that are filed).  I think this would be better phrased as "Database %q shares bucket %q with databases %s. This may result in unexpected behaviour if security is not defined consistently."

Reply:
Makes more sense; this is really a cool warning message. Changes are committed to the latest revision.